react-app/ccf/src/pages/settings/settings.tsx

```diff
@@ -49,13 +88,83 @@ function AccountSettingsPage(): JSX.Element {
     }
   };
 
-  const handleSubmit = (e: any) => {
+  const handleSubmit = async (e: React.FormEvent) => {
     e.preventDefault();
-    if (!specialChar || !capitalLetter || !number || pwdUnmatched || !currentPasswordMatched) {
+    setUpdateError(null);
+    setUpdateSuccess(false);
+
+    if (!specialChar || !capitalLetter || !number || pwdUnmatched) {
       console.log("Failed to submit. One requirement was not met.");
       return;
     }
-    console.log("Password changed successfully:", pwd);
+
+    const auth = getAuth();
```

Comment:
Nice work, just make sure to import the shared auth object from index.tsx rather than getting it again.